File: src/main/scala/org/raisercostin/jedi/StreamLocation.scala
package org.raisercostin.jedi

import java.io.File
import java.io.InputStream

import scala.language.implicitConversions
import scala.language.reflectiveCalls
import scala.util.Try


case class StreamLocation(val inputStream: InputStream) extends InputLocation with IsFile{
  def exists: Boolean = true
  def nameAndBefore: String = inputStream.toString()
  def raw = "inputStream[" + inputStream + "]"
  override def unsafeToInputStream: InputStream = inputStream
  override def size:Long = ???
  def metaLocation:Try[NavigableInOutLocation/*MetaRepr*/] = ???
}

case class StreamProviderLocation(inputStream: ()=>InputStream) extends InputLocation with IsFile{
  def exists: Boolean = true
  def nameAndBefore: String = inputStream.toString()
  def raw = "inputStream[" + inputStream + "]"
  override def unsafeToInputStream: InputStream = inputStream.apply()
  override def size:Long = ???
  def metaLocation:Try[NavigableInOutLocation/*MetaRepr*/] = ???
}

File: src/main/kotlin/org/raisercostin/jedi/StreamLocation.kt
package org.raisercostin.jedi

import java.io.File
import java.io.InputStream

import scala.language.implicitConversions
import scala.language.reflectiveCalls
import scala.util.Try


data class StreamLocation(val inputStream: InputStream) : InputLocation , IsFile{
  fun exists: Boolean = true
  fun nameAndBefore: String = inputStream.toString()
  fun raw ()= "inputStream<" + inputStream + ">"
  override fun unsafeToInputStream: InputStream = inputStream
  override fun size:Long = ???
  fun metaLocation:Try<NavigableInOutLocation/*MetaRepr*/> = ???
}

data class StreamProviderLocation(inputStream: ()->InputStream) : InputLocation , IsFile{
  fun exists: Boolean = true
  fun nameAndBefore: String = inputStream.toString()
  fun raw ()= "inputStream<" + inputStream + ">"
  override fun unsafeToInputStream: InputStream = inputStream.apply()
  override fun size:Long = ???
  fun metaLocation:Try<NavigableInOutLocation/*MetaRepr*/> = ???
}